File: Python/Contagem-de-permutações.py
def count_permutations(s:str):
    valores_unicos:set = set(list(s))
    repeticao = [s.count(char) for char in valores_unicos]
    num_permutacao:int = 0

    def fatorial(num):
        if num in [0, 1]:
            return 1
        else:
            fat = 1
            for val in range(1, num + 1):
                fat *= val
            return fat

    if repeticao:
        num_permutacao = fatorial(len(s))
        dividendo = 1
        for val in repeticao:
            dividendo *= fatorial(val)
    return int(num_permutacao / dividendo)

count_permutations("abb")
count_permutations("abc")
count_permutations("racecar")
count_permutations("freecodecamp")

File: Javascript/Contagem-de-permutações.js
function fatorial(num)
{
  let fat = 1;
  if(num === 0)
  {
    return 1;
  }
  else if(num === 1)
  {
    return 1;
  }
  else
  {
    for(let i = 1; i <= num; i++)
    {
      fat = fat * i;
    }
    return fat;
  }
}

function countCharacter(str, strSet)
{
  let contagem = [];
  let cont = 0;
  
  for(var item of strSet)
  {
    for(let i = 0; i < str.length; i++)
    {
        if (item === str[i])
        {
          cont += 1;
        }
    }
    contagem.push(cont);
    cont = 0;
  }
  
  return contagem;
}

function countPermutations(str) 
{
  const meuSet = new Set(str);
  const repeticao = countCharacter(str, meuSet);
  let dividendo = 1;
  
  for(let i = 0; i < repeticao.length; i++)
  {
    dividendo *= fatorial(repeticao[i]);
  }
 
  console.log(dividendo);
  return fatorial(str.length) / dividendo;
}
